packages/newspringchurchapp/src/ui/TileImage/index.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { View } from 'react-native';
+import PropTypes from 'prop-types';
+import { pure, compose } from 'recompose';
+import { H5, styled, withTheme } from '@apollosproject/ui-kit';
+import GradientOverlayImage from '../GradientOverlayImage';
+
+const CardView = styled(
+  ({ theme }) => ({
+    borderRadius: theme.sizing.baseUnit,
+    overflow: 'hidden',
+    width: '100%',
+    height: '100%',
+    aspectRatio: 1,
+  }),
+  'TileImage'
+)(View);
+
+const Title = styled(
+  ({ theme }) => ({
+    position: 'absolute',
+    bottom: theme.sizing.baseUnit,
+    left: theme.sizing.baseUnit,
+    right: theme.sizing.baseUnit,
+    backgroundColor: theme.colors.transparent,
+    color: theme.colors.lightPrimary,
+  }),
+  'TileImage.Text'
+)(H5);
+
+const SquareGradientOverlayImage = styled({
+  aspectRatio: 1,
+})(GradientOverlayImage);
+
+const enhance = compose(
+  withTheme(({ theme: { colors } = {} } = {}) => ({ theme: { colors } })),
+  pure
+);
+
+const TileImage = enhance(({ image, isLoading, overlayColor, text }) => (
+  <CardView>
+    <SquareGradientOverlayImage
+      source={image}
+      isLoading={isLoading}
+      maintainAspectRatio={false}
+      overlayColor={overlayColor || null}
+    />
+    <Title>{text}</Title>
+  </CardView>
+));
+
+TileImage.propTypes = {
+  image: GradientOverlayImage.propTypes.source,
+  isLoading: PropTypes.bool,
+  link: PropTypes.string,
+  onPressItem: PropTypes.func,
+  overlayColor: PropTypes.arrayOf(PropTypes.string),
+  text: PropTypes.string,
+};
+
+TileImage.defaultProps = {
```

Comment:
Why is this needed?

Reply:
Not sure. This was copied over from TAP core. Maybe just to make sure that the text doesn't get accidentally set to something else?

Reply:
/shrug